### PR TITLE
(#917) Improve performances of CollectionEnvelope equals

### PR DESCRIPTION
**Pull request by victornoel**
This is for #917, there is not much in it, I also added a test which, even if not really needed in relation to the change, is not useless :)

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1087](https://github.com/yegor256/cactoos/issues/1087) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1087](https://github.com/yegor256/cactoos/issues/1087) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/e18e8823-633f-43a9-81e5-b7349d0fe7f7); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by victornoel** [comment]
@llorllale this PR is so small that I think you can directly review it, I don't think it is a good idea that I do it ;) (well, I believe it's good to merge :).

**Comment by llorllale** [comment]
@0crat refuse

File: src/test/java/org/cactoos/collection/CollectionEnvelopeTest.java [hunk 1]
```diff
@@ -85,6 +85,16 @@ public void notEqualToCollectionOfDifferentElements() {
         );
     }
 
+    @Test
+    public void equalToItself() {
+        final CollectionOf<String> col = new CollectionOf<>("val1", "val2");
+        MatcherAssert.assertThat(
```
Comment (llorllale) [inline src/test/java/org/cactoos/collection/CollectionEnvelopeTest.java]: @victornoel let's use `Assertion`.

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/1087#issuecomment-478183469))

@llorllale The user @victornoel[/z](https://www.0crat.com/u/victornoel) resigned from [#1087](https://github.com/yegor256/cactoos/issues/1087), please stop working. Reason for job resignation: Order was cancelled

**Comment by llorllale** [comment]
@victornoel 1 comment

**Comment by victornoel** [comment]
@llorllale oups, good point, I've updated the PR

**Comment by 0crat** [comment]
This pull request [#1087](https://github.com/yegor256/cactoos/issues/1087) is assigned to @atapin[/z](https://www.0crat.com/u/atapin), here is [why](https://www.0crat.com/footprint/C63314D6Z/b5076553-bc08-4744-9854-66f1b9787bb4); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by victornoel** [comment]
@llorllale ping

**Comment by llorllale** [comment]
@0crat refuse

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/yegor256/cactoos/issues/1087#issuecomment-478802616))

@llorllale The user @atapin[/z](https://www.0crat.com/u/atapin) resigned from [#1087](https://github.com/yegor256/cactoos/issues/1087), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#1087` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#1087](https://github.com/yegor256/cactoos/issues/1087) is now out of scope

**Pull request merged**